src/main/java/seedu/address/logic/parser/SortCommandParser.java

```diff
@@ -64,11 +72,15 @@ private SortArgument convertArguments(PrefixArgument prefixArg) {
         // Sorting is reversed if the prefix is followed by a '!'
         boolean isReverse = !arg.isEmpty() && arg.charAt(0) == '!';
 
-        if (prefix.equals(PREFIX_TAG)) {
-            Tag tag = isReverse ? new Tag(arg.substring(1)) : new Tag(arg);
-            return new SortArgument(prefix, isReverse, tag);
-        } else {
-            return new SortArgument(prefix, isReverse, null);
+        try {
+            if (prefix.equals(PREFIX_TAG)) {
+                Tag tag = isReverse ? parseTag(arg.substring(1)) : parseTag(arg);
+                return new SortArgument(prefix, isReverse, tag);
+            } else {
+                return new SortArgument(prefix, isReverse, null);
+            }
+        } catch (ParseException e) {
+            return null;
```

Comment:
If I recall correctly, returning `null` is not a good practice.

Reply:
Both `convertArguments`, and `convertPrefixArgToSortArg` that calls it are private methods. 
Hence, there is no risk of the null objects returned triggering a `NullPointerException` outside of this class. 

As for the `parse` method, right after `convertPrefixArgToSortArg` is called, the returned list is checked for `null` objects.
Thus, the `null` here should be safe to use.

Reply:
Ok